smsdk/client_v0.py

```diff
@@ -116,13 +119,30 @@ def __init__(self, tenant, site_domain="sightmachine.io"):
             a non-standard environment.
         :type site_domain: :class:`string`
         """
+        # Parse the input string
+        url = tenant
+        parsed_uri = urlparse(url)
+
+        # Extract tenant from the path
+        tenant = (
+            parsed_uri.netloc.split(".", 1)[0]
+            if parsed_uri.netloc
+            else url.split(".", 1)[0]
+        )
 
-        self.tenant = tenant
+        # Extract protocol
+        if parsed_uri.scheme:
+            protocol = parsed_uri.scheme
 
-        # Handle internal configuration
-        self.config = {}
-        self.config["protocol"] = "https"
-        self.config["site.domain"] = site_domain
+        # Extract site domain
+        if parsed_uri.netloc:
+            site_domain = parsed_uri.netloc.replace(f"{tenant}.", "")
+        else:
+            if f"{tenant}." in url:
+                site_domain = url.replace(f"{tenant}.", "", 1)
+
+        self.tenant = tenant
+        self.config = {"protocol": protocol, "site.domain": site_domain}
```

Comment:
Unit tests for these changes?

Looks like everyone calls the class as:
```python
dt = Client("demo-sdk-test")
```

```python
def test_uri():
   # Should this complain?
    dt = Client("")

   # Should this complain?
    dt = Client("demo-sdk-test.localnet")
    assert dt.tenant == "demo-sdk"  # ???
    assert dt.config["site.domain"] == "sightmachine.io"

    dt = Client("://demo-sdk-test.localnet")
    assert dt.tenant == "demo-sdk"  # ???
    assert dt.config["site.domain"] == "sightmachine.io"

    dt = Client("https://demo-sdk-test")
    assert dt.tenant == "demo-sdk"
    assert dt.config["site.domain"] == "sightmachine.io"
    assert dt.config["protocol"] == "https"

    dt = Client("https://demo-sdk-test.localnet")
    assert dt.tenant == "demo-sdk"
    assert dt.config["site.domain"] == "localhost"
    assert dt.config["protocol"] == "http"

```

Reply:
Added a new unit test and also handling all the  above mentioned special cases